src/sentry/monitors/utils.py

```diff
@@ -41,6 +43,91 @@ def valid_duration(duration: Optional[int]) -> bool:
     return True
 
 
+# Returns serializer appropriate group_ids corresponding with check-in trace ids
+def fetch_associated_groups(
+    trace_ids: List[str], organization_id: int, project_id: int, start: datetime, end
+) -> Dict[str, Set[int]]:
+    from snuba_sdk import (
+        Column,
+        Condition,
+        Direction,
+        Entity,
+        Limit,
+        Offset,
+        Op,
+        OrderBy,
+        Query,
+        Request,
+    )
+
+    from sentry.eventstore.base import EventStorage
+    from sentry.eventstore.snuba.backend import DEFAULT_LIMIT, DEFAULT_OFFSET
+    from sentry.snuba.dataset import Dataset
+    from sentry.snuba.events import Columns
+    from sentry.utils.snuba import DATASETS, raw_snql_query
+
+    dataset = Dataset.Events
+
+    # add 30 minutes on each end to be safe
+    query_start = start - timedelta(minutes=30)
+    query_end = end + timedelta(minutes=30)
+
+    cols = [col.value.event_name for col in EventStorage.minimal_columns[dataset]]
+    cols.append(Columns.TRACE_ID.value.event_name)
+
+    # query snuba for related errors and their associated issues
+    snql_request = Request(
+        dataset=dataset.value,
+        app_id="eventstore",
+        query=Query(
+            match=Entity(dataset.value),
+            select=[Column(col) for col in cols],
+            where=[
+                Condition(
+                    Column(DATASETS[dataset][Columns.TIMESTAMP.value.alias]),
+                    Op.GTE,
+                    query_start,
+                ),
+                Condition(
+                    Column(DATASETS[dataset][Columns.TIMESTAMP.value.alias]),
+                    Op.LT,
+                    query_end,
+                ),
+                Condition(
+                    Column(DATASETS[dataset][Columns.TRACE_ID.value.alias]),
+                    Op.IN,
+                    trace_ids,
+                ),
+                Condition(
+                    Column(DATASETS[dataset][Columns.PROJECT_ID.value.alias]),
+                    Op.EQ,
+                    project_id,
+                ),
+            ],
+            orderby=[
+                OrderBy(Column(DATASETS[dataset][Columns.TIMESTAMP.value.alias]), Direction.DESC),
+            ],
+            limit=Limit(DEFAULT_LIMIT),
+            offset=Offset(DEFAULT_OFFSET),
+        ),
+        tenant_ids={
+            "referrer": "api.serializer.checkins.trace-ids",
+            "organization_id": organization_id,
+        },
+    )
+
+    trace_groups: Dict[str, Set[int]] = defaultdict(set)
+
+    result = raw_snql_query(snql_request, "api.serializer.checkins.trace-ids", use_cache=False)
+    if "error" not in result:
```

Comment:
If there is an error is this just returning an empty list then?

Reply:
empty list

Reply:
v minor nit: Maybe better to be explicit with `if 'error' in result: return []` type thing